internal/extproc/messages_processor_test.go

```diff
@@ -826,6 +826,84 @@ func TestMessagesProcessorUpstreamFilter_ProcessRequestHeaders_WithHeaderMutatio
 		// Check that original headers remain unchanged.
 		require.Equal(t, "bearer token123", headers["authorization"])
 	})
+
+	t.Run("multiple header mutations with same key - last one wins", func(t *testing.T) {
```

Comment:
My implementation relies on the ( current ) behaviour that when setting the same multiple headers with the same keys the last one wins. 

I have added this test to ensure that behaviour in the future isn't broken ( by accident )